code/modules/mob/living/simple_animal/friendly/drone/_drone.dm

```diff
@@ -273,3 +276,10 @@
 
 /mob/living/simple_animal/drone/electrocute_act(shock_damage, obj/source, siemens_coeff = 1, safety = 0, tesla_shock = 0, illusion = 0, stun = TRUE)
 	return 0 //So they don't die trying to fix wiring
+
+/mob/living/simple_animal/drone/can_speak_in_language(datum/language/dt)
```

Comment:
not just in the proc
around all calls to the proc
print the type of the mob
etc.

world << "CALLING:"
call()
world << "CALLED"
if nothing is inbetween, look elsewhere etc. 

I'm tired so I cant really help further, sorry.